### PR TITLE
Fix for task id as valid UUID

### PR DESCRIPTION
**Pull request by Rieven**
### Changes
- When fetching a task by id, we do not check if the string is a valid UUID. 
- check POST task id is a valid UUID

### Issue link
https://github.com/minvws/nl-kat-coordination/issues/3743

Closes https://github.com/minvws/nl-kat-coordination/issues/3743

### Demo

image

### QA notes

_Please add some information for QA on how to test the newly created code._

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/rocky/views/task_detail.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def get_context_data(self, **kwargs):
 class NormalizerTaskJSONView(TaskDetailView):
     task_type = "normalizer"
     plugin_type = "normalizer"
+    template_name = "tasks/normalizer_task_detail.html"
```
Comment (Rieven) [inline rocky/rocky/views/task_detail.py:60]: My test failed on this part, because it was missing the template name

**Review by underdarknl — DISMISSED** [review]

File: rocky/tests/scheduler/test_scheduler_errors.py [hunk 2]
```diff
@@ -37,3 +38,27 @@ def test_tasks_view_too_many_requests_error(rf, client_member, mock_scheduler):
...
+
+
+def test_reschedule_task_bad_task_id(rf, client_member, mock_bytes_client, mock_scheduler):
+    request = setup_request(
+        rf.post("task_list", {"action": "reschedule_task", "task_id": "/delete"}), client_member.user
+    )
+
+    response = BoefjesTaskListView.as_view()(request, organization_code=client_member.organization.code)
+
+    assert response.status_code == 200
```
Comment (underdarknl) [inline rocky/tests/scheduler/test_scheduler_errors.py:62]: This should probably return a 404, not a 200.

**Review by ammar92 — APPROVED** [review]

File: rocky/rocky/views/scheduler.py [hunk 3]
```diff
@@ -112,13 +112,15 @@ def get_report_child_name_form(self):
...
-        task = self.scheduler_client.get_task_details(task_id)
-
-        if task.organization_id() != self.organization.code:
-            messages.error(self.request, SchedulerTaskNotFound.message)
-            return None
+        try:
+            task = self.scheduler_client.get_task_details(task_id)
+            if task.organization_id() != self.organization.code:
+                messages.error(self.request, SchedulerTaskNotFound.message)
+                return None
```
Comment (underdarknl) [inline rocky/rocky/views/scheduler.py:119]: Shouldn't this also be :
raise Http404()

File: rocky/rocky/views/scheduler.py [hunk 4]
```diff
@@ -169,19 +171,16 @@ def get_output_oois(self, task):
...
-                    {
-                        "oois": self.get_output_oois(task),
-                        "valid_time": task.data.raw_data.boefje_meta.ended_at.strftime("%Y-%m-%dT%H:%M:%S"),
-                    },
-                    safe=False,
-                )
-            return task
-        except SchedulerError as error:
-            return messages.error(self.request, error.message)
+        task = self.get_task_details(self.kwargs["task_id"])
```
Comment (underdarknl) [inline rocky/rocky/views/scheduler.py:174]: This flow could also reaise a SchedulerTaskNotFound, right?

File: rocky/rocky/views/scheduler.py [hunk 5]
```diff
@@ -210,18 +209,19 @@ def schedule_task(self, task: Task) -> None:
...
-        task = self.scheduler_client.get_task_details(task_id)
+        task = self.get_task_details(task_id)
 
-        if task.organization_id() != self.organization.code:
-            messages.error(self.request, SchedulerTaskNotFound.message)
-            return
+        if task is not None:
+            if task.organization_id() != self.organization.code:
+                messages.error(self.request, SchedulerTaskNotFound.message)
+                return
```
Comment (underdarknl) [inline rocky/rocky/views/scheduler.py:217]: raise Http404()

**Comment by dekkers** [comment]
> > ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=minvws_nl-kat-coordination&pullRequest=3744) **Quality Gate failed**
> > Failed conditions  [70.6% Coverage on New Code](https://sonarcloud.io/component_measures?id=minvws_nl-kat-coordination&pullRequest=3744&metric=new_coverage&view=list) (required ≥ 80%)
> > [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=minvws_nl-kat-coordination&pullRequest=3744)
> 
> We should exclude tests from covarage

We shouldn't and that isn't a problem here. The test code has 100% coverage, the code that should be tested is missing coverage:

image

**Comment by Rieven** [comment]
> > > ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=minvws_nl-kat-coordination&pullRequest=3744) **Quality Gate failed**
> > > Failed conditions  [70.6% Coverage on New Code](https://sonarcloud.io/component_measures?id=minvws_nl-kat-coordination&pullRequest=3744&metric=new_coverage&view=list) (required ≥ 80%)
> > > [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=minvws_nl-kat-coordination&pullRequest=3744)
> > 
> > 
> > We should exclude tests from covarage
> 
> We shouldn't and that isn't a problem here. The test code has 100% coverage, the code that should be tested is missing coverage:
> 
> image

I was a bit confused that lines to cover is also counted for test files, but may has nothing to do with the covarage calculations?
image

**Review by underdarknl — APPROVED** [review]

**Pull request merged**